visualization/dashboard/src/MLIDashboard/Controls/CohortEditor/CohortEditor.tsx

```diff
@@ -328,21 +372,21 @@ export class CohortEditor extends React.PureComponent<ICohortEditorProps, ICohor
     }
 
     private updateFilter(filter: IFilter): void {
-        const isEditingFilter = this.state.isEditingFilter;
         let filters = [...this.state.filters];
-        if (!isEditingFilter) {
+
+        if (this.state.openedFilter.column!== this.state.editingFilterIndex) {
```

Comment:
I'm not sure what is happening here, I think the need for 'push' case could be done away with, by setting state.filterIndex to this.state.filters.length in the case of creating new filter, and to the index of the exisiting filter if being edited. Is there some alternative behavior, like if the user presses edit filter, then changes the left column selection, to not treat it as an edit but rather some other change?

Reply:
I'm not sure if I understand this correctly. Can you elaborate on the alternative behavior. Also I'm not sure about state.filterIndex implementation to replace push so let me get back on that.

Reply:
Code now adds element at index:  this.state.filterIndex which is either set to filter index if editing or filter.length when adding a new filter.